backend/api/views/TimeRecordView.py

```diff
@@ -0,0 +1,47 @@
+from rest_framework import viewsets
+from rest_framework.permissions import IsAuthenticated
+from api.models import TimeRecord, EventType
+from api.serializers.TimeRecordSerializer import TimeRecordSerializer
+from drf_spectacular.utils import extend_schema, OpenApiParameter
+from django.db.models import Q
+from django.db.models.functions import Cast
+from django.db.models.functions import Collate
+from rest_framework.exceptions import ValidationError
+from rest_framework import status
+
+import logging
+
+logger = logging.getLogger('django')
+
+
+@extend_schema(tags=['Time Record'], request=TimeRecordSerializer)
+class TimeRecordViewSet(viewsets.ModelViewSet):
+    queryset = TimeRecord.objects.all()
+    serializer_class = TimeRecordSerializer
+    permission_classes = [IsAuthenticated]
+    http_method_names = ['get', 'post', 'delete', 'patch']
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+
+        athlete_name = self.request.query_params.get('athlete_name', None)
+        event_type_id = self.request.query_params.get('event_type_id', None)
+
+        if athlete_name:
```

Comment:
When I search using the first name:

![image](https://github.com/MarisiaS/SMM/assets/95152310/5d282426-9a27-47b6-928e-b68620d73fc1)

When I search using the last name:
![image](https://github.com/MarisiaS/SMM/assets/95152310/7179fc83-b4ab-4c32-b4a9-ded1435e8e3e)

When I search using the full name:
![image](https://github.com/MarisiaS/SMM/assets/95152310/08f6d33c-acbe-4ed1-84b8-6cfa366faba5)


I found this [article](https://stackoverflow.com/questions/50682544/how-to-concatenate-two-model-fields-in-a-django-queryset) about concatenating fields.

In our case I think we need to add the collation as well. So something like this:
```
from django.db.models.functions import Collate, Concat
from django.db.models import Q, F, Value

queryset = queryset.annotate(
            full_name_deterministic=Collate(
                Concat(F('athlete__first_name'), Value(' '), F('athlete__last_name')),
                'und-x-icu'
            )
```

I believe adding this is crucial, as my initial thought when considering filtering by name was to use the full name.

Reply:
Also, maybe it is better to use `istartswith` to make it case insensitive.

Reply:
@MarisiaS 
I made those changes but if we leave istartswith, the search would always start with the athlete's first name in order to get the results. Is that ok?